src/es6/SirenAction.js

```diff
@@ -146,6 +146,10 @@ const _performSirenAction = function(action, fields, tokenValue) {
 				if (!entity) {
 					return window.D2L.Siren.EntityStore.remove(url.href, token);
 				}
+				const selfLink = entity.getLinkByRel('self');
+				if (selfLink && selfLink.href !== url.href) {
+					return window.D2L.Siren.EntityStore.update(selfLink.href, token, entity);
+				}
 				return window.D2L.Siren.EntityStore.update(url.href, token, entity);
```

Comment:
To maintain the existing behavior, it's possible someone is using this and returning an entity missing a `self` href, so I will update the store with the returned entity and original request href if the self href is missing.